src/components/TrackTile/ManageTrackers/ManageTrackerRow.tsx

```diff
@@ -1,41 +1,95 @@
 } from 'react-native';
 import { t } from '../../../../lib/i18n';
 import { Tracker } from '../services/TrackTileService';
 import Indicator from '../icons/indicator';
 import { Text } from '../styles/Text';
 import { tID } from '../common/testID';
 import { useIcons, createStyles } from '../../BrandConfigProvider';
-import { useStyles } from '../../../hooks';
+import { useStyles, useTheme } from '../../../hooks';
+import { IconButton } from 'react-native-paper';
+import { Easing } from 'react-native-reanimated';
 
-type Props = TouchableHighlightProps & {
+type Props = Omit<TouchableHighlightProps, 'onPress' | 'onPressIn'> & {
   tracker: Tracker;
   isBeingDragged?: boolean;
-  isDraggable?: boolean;
+  isEditable?: boolean;
   endAdornment?: React.ReactNode;
+  onToggleInstall?: () => void;
+  onDrag?: () => void;
+  onOpen?: () => void;
 };
 
 export const ManageTrackerRow: FC<Props> = (props) => {
-  const { tracker, endAdornment, ...remainingProps } = props;
-  const { isBeingDragged, isDraggable, ...highlightProps } = remainingProps;
+  const { tracker, endAdornment, onToggleInstall, ...remainingProps } = props;
+  const { isBeingDragged, isEditable, onDrag, onOpen, ...highlightProps } =
+    remainingProps;
   const { styles } = useStyles(defaultStyles);
-  const inactive = !tracker.metricId;
+  const theme = useTheme();
   const id = tracker.metricId || tracker.id;
-  const { Menu, ChevronRight, ChevronLeft } = useIcons();
+  const { Menu, ChevronRight, ChevronLeft, Circle, CheckCircle } = useIcons();
+  const animation = useRef(new Animated.Value(0)).current;
+
+  useEffect(() => {
+    Animated.timing(animation, {
+      toValue: !isEditable ? 0 : 1,
+      duration: 400,
+      useNativeDriver: false,
+      easing: isEditable ? Easing.linear : Easing.in(Easing.linear),
+    }).start();
+  }, [isEditable, animation]);
 
   return (
-    <TouchableHighlight {...highlightProps}>
+    <TouchableHighlight
+      onPress={isEditable ? undefined : onOpen}
+      {...highlightProps}
+    >
       <>
         {isBeingDragged && <View style={styles.divider} />}
         <View
           style={[styles.container, isBeingDragged && styles.containerActive]}
         >
-          <View style={[styles.icon, inactive && styles.iconInactive]}>
+          <Animated.View
+            style={[
+              styles.startAdornmentView,
+              {
+                transform: [
+                  {
+                    translateX: animation.interpolate({
+                      inputRange: [0, 1],
+                      outputRange: [-65, 0],
+                    }),
+                  },
+                ],
+                width: animation.interpolate({
+                  inputRange: [0, 1],
+                  outputRange: [0, 24],
+                }),
+              },
+            ]}
+            testID={tID(`toggle-button-${id}`)}
+          >
+            <IconButton
+              onPress={onToggleInstall}
+              icon={() =>
```

Comment:
I've been seeing this warning (`Do not define components during render.`) a lot more lately.  It kind of forces you to create a component, with a parameter that `tracker.installed` can feed.  I've also been in a situation where I couldn't find a way _out_ of this warning no matter what I did (home screen refresh button).  I think we should definitely let it help nudge us towards preferring components/memos over inline func components like this, though.

Reply:
To address I've been wrapping the components with useCallback which suppresses the warning.